### PR TITLE
Adding first commit of the android webview shim

### PR DESCRIPTION
**Pull request by gauntface**
cc @addyosmani @agektmr @zkoch

Thoughts on this?

I've tested locally, although I would like to get some saucelabs tests added in ASAP which can be done with Android emulator + Android browser.

Sadly, because payment is mobile only, it's going to be hard to test on actual devices, I've only managed to get Appium to work with Chrome stable on Android.

### CODE REVIEW

**Comment by zkoch** [comment]
This looks good to me. Question: is that webview check enough? Looking at the [possible strings](https://developer.chrome.com/multidevice/user-agent), it seems like we might have to do more for pre-Lollipop? Not sure if we need to be worried about those...

**Comment by gauntface** [comment]
@zkoch my reasoning for this check is the following:

- Pre Android-K the webview was the old school WebKit version - No PaymentRequest API.
- Android-K Has the Chrome WebView but it's a fixed version - no updating and it doesn't have the PaymentRequest API.
- Android-L and above has the auto-updating webview.
    - In M53 and below - the PaymentRequest API was not available.
    - In M54 the PaymentRequest API was available when it doesn't work
    - In M55 and above - The PaymentRequest API should be removed in the WebView, meaning this shim is no longer required.

All of the above means, we need to only look for the auto-updating WebView version 54.

Is any of that thought process wrong?

**Comment by zkoch** [comment]
> Is any of that thought process wrong?

Nope! SGTM. 🎆

**Comment by Edouard-chin** [comment]
Thanks guys 😍

**Comment by agektmr** [comment]
@gauntface @zkoch Chrome started to support PR API since M53 FWIW.

File: src/interventions/android-webview-54.js [hunk 1]
```diff
@@ -42,6 +42,6 @@ module.exports = (window, navigator) => {
 
   const regexResult = regexCheck.exec(userAgent);
   if (regexResult !== null) {
-    delete window.PaymentRequest;
+    window.PaymentRequest = null;
```
Comment (agektmr) [inline src/interventions/android-webview-54.js:45]: I'd set it to `undefined` just in case.

Comment (gauntface) [reply src/interventions/android-webview-54.js:45]: @addyosmani was recommending against this.

**Comment by gauntface** [comment]
@agektmr @zkoch Do you know if the PaymentRequest API was enabled in WebView M53? I'm more than happy to add that to this intervention is desired.

I also noticed that PaymentAddress was available, I've left it on the window untouched - should I remove that from the window as well?

File: src/interventions/android-webview-54.js (hunk 1 above)
Comment (addyosmani) [reply src/interventions/android-webview-54.js:45]: The `delete` keyword leads to negative performance effects with V8's hidden classes and it's generally recommended to just set to `null` instead.  I would encourage setting the object to `null` here, in particular as you're working with the `window` object which will likely be regularly used during the page's lifecycle.

**Review by addyosmani — APPROVED** [review]
LGTM with the `null` change in place.

**Comment by agektmr** [comment]
@gauntface PR API is enabled since M53 so you should take care of that.
RE: `PaymentAddress`, there shouldn't be any harm, but might worth disabling it just in case.

**Comment by gauntface** [comment]
@zkoch @agektmr Ok so the regex will now capture both M53 and M54 user agents and set both PaymentRequest and PaymentAddress to null.

Does this seem like a safe change for you?

**Comment by zkoch** [comment]
Seems safe to me! Thanks! (sorry for missing the 53 thing the first time, doh!)

File: src/interventions/android-webview-54.js (hunk 1 above)
Comment (madmath) [reply src/interventions/android-webview-54.js:45]: Hi, 

The WooCommerce impl of PaymentRequest checks for PR support by doing 

'PaymentRequest' in window

which would imply that this solution is not sufficient (that check will still be true). I would keep the "delete window.PaymentRequest" solution.

WooCommerce link: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/assets/js/payment-request.js#L31

**Pull request merged**